.clang-format

```diff
@@ -0,0 +1,148 @@
+---
+# this file work is a derivative of .clang-format which follows
+# https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md
+#
+Language:        Cpp
+BasedOnStyle:  Microsoft
```

Comment:
Since this config is based on the `Microsoft` style, a lot of the options listed below are actually already properly configured and should be removed...

Reply:
Now it's a very short list. Do I need to remove Microsoft?

Reply:
@paulcam206 ?

Reply:
> Now it's a very short list. Do I need to remove Microsoft?

I don't understand -- the list should be short... the `BasedOnStyle: Microsoft` directive sets a bunch of presets for what the `clang` community considers to be our coding style standards. Other directives in this file should be modifications to fit our actual standards. :)